OrbitCaptureClient/include/OrbitCaptureClient/GpuQueueSubmissionProcessor.h

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_CAPTURE_CLIENT_GPU_QUEUE_SUBMISSION_PROCESSOR_H_
+#define ORBIT_CAPTURE_CLIENT_GPU_QUEUE_SUBMISSION_PROCESSOR_H_
+
+#include <absl/container/flat_hash_map.h>
+
+#include <algorithm>
+#include <functional>
+#include <string>
+#include <vector>
+
+#include "capture.pb.h"
+#include "capture_data.pb.h"
+
+// `GpuQueueSubmission` proto objects come with GPU timestamps (rather than CPU timestamps) for the
+// command buffer and debug marker timings. In order convert those timestamps into CPU time, each
+// `GpuQueueSubmission` contains a timestamp before (pre) and after (post) the vkQueueSubmit driver
+// call. For the driver we already have timestamps in the `GpuJob` events. Together with the thread
+// id, this allows us to establish a 1:1 mapping of `GpuJob`s and `GpuQueueSubmission.
+//
+// This class allows to convert `GpuQueueSubmission` events (with GPU timestmaps) to command buffer
+// and debug marker `TimerInfo`s (with CPU timestamps). Thereby, it manages the mapping from
```

Comment:
I think "thereby" is not fully appropriate here. "For that"?

Reply:
Done.